### ISSUE TITLE
FileSystem.documentDirectory changed between SDK 32 and 35

### ISSUE THREAD

**Issue by CampbellMG** [expo/expo#5942]
### Environment

 Expo CLI 3.2.3 environment info:
    System:
      OS: Windows 10
    Binaries:
      npm: 6.9.0 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.5.0.0 AI-191.8026.42.35.5791312

This issue occurs on both iOS and Android standalone apps when comparing SDK 32 to 35.

### Steps to Reproduce

1. Create an empty project targeting SDK 32
2. Add a single text element displaying FileSystem.documentDirectory
3. Generate a standalone IPA or APK
4. Update to SDK 35
5. Replace FileSystem import with modular imports from 'expo-file-system'
6. Generate a standalone IPA or APK

### Expected Behavior

The document directory should remain the same between SDK versions

### Actual Behavior

The document directory has lost the last two elements as seen below. 

This is causing the RCTAsyncLocalStorage to appear in a different location meaning all AsyncStorage elements are gone. The values are still in the same directory but Expo is no longer looking in the original location. 

Additionally, any files stored using this variable are now inaccessible without appending the difference to documentsDirectory.

SDK 32 - `Alert.alert('Document Directory', FileSystem.documentDirectory)`

![SDK 32 - `Alert.alert('Document Directory', FileSystem.documentDirectory)`](https://i.imgur.com/0dS787U.png)

SDK 35 - `Alert.alert('Document Directory', FileSystem.documentDirectory)`

![SDK 35 - `Alert.alert('Document Directory', FileSystem.documentDirectory)`](https://i.imgur.com/83oI5Hw.png)

### Reproducible Demo

This is hard to reproduce online as it involves generating standalone versions of the app but I have APK's demonstrating this issue that I can provide if necessary.

**Comment by CampbellMG** [comment]
I have found the issue causing this problem, it's intentional but leaves a question remaining. 

In host.exp.exponent.utils.ScopedContext, it seems the file directory changed in 33 and a migration was in place for version 33 and 34 but dropped for 35. 

So my question is, how is this migration expected to work?
Does Expo only support migrations one SDK at a time?

**Comment by cruzach** [comment]
The migration code was dropped since SDK 32 was deprecated in this release, although it could've been communicated better that if you're using SDK 32, you **have** to upgrade to SDK 33 and/or SDK 34 before upgrading to SDK 35. But yeah, this is one of the reasons we recommend upgrading one SDK at a time

**Comment by mojjy** [comment]
Hi

I upgraded my sdk version to 35, from an earlier version, some users lost data, and then I realised that I had not upgraded first to 34.  The next day I released a new version, with sdk 34.  Is there any way I can recover the data for the users who downloaded the earlier version. (This is on ios) 

When can I upgrade to 35, do I need to wait for every user to first upgrade their apps. How will I know this?
